vaadin-dashboard-flow-parent/vaadin-dashboard-flow/src/main/java/com/vaadin/flow/component/dashboard/Dashboard.java

```diff
@@ -64,6 +64,9 @@
 @NpmPackage(value = "@vaadin/dashboard", version = "24.7.0-alpha2")
 public class Dashboard extends Component implements HasWidgets, HasSize {
 
+    private static final ThreadLocal<Boolean> suppressClientUpdates = ThreadLocal
```

Comment:
I'm not sure why this can't just be a boolean instance property. Could you explain please?

Reply:
Technically it wouldn't be thread-safe otherwise. The component could be processing a request from the client in one thread which sets the boolean flag to `true`. Meanwhile another thread could make modifications to the component, which would then fail to sync to the client. Not sure if it's a big concern, but it seemed the correct way of doing it.